pkg/storegateway/indexheader/stream_binary_reader.go

```diff
@@ -148,7 +163,96 @@ func newFileStreamBinaryReader(path string, postingOffsetsInMemSampling int, log
 		return nil, err
 	}
 
-	return r, nil
+	return r, err
+}
+
+// restoreSampleFromHeader loads in sampled index-header from disk.
+func (r *StreamBinaryReader) restoreSampleFromHeader(logger log.Logger, samplesPath string, sampleData []byte, postingOffsetsInMemSampling int) (err error) {
```

Comment:
This method seems like it's doing the opposite of its name: It's loading index header data from a sampled version on disk.

```suggestion
func (r *StreamBinaryReader) loadFromSampledHeader(logger log.Logger, samplesPath string, sampleData []byte, postingOffsetsInMemSampling int) (err error) {
```